test/integration/middleware_test.rb

```diff
@@ -22,4 +22,28 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
       Rails.configuration.webpack.dev_server.enabled = false
     end
   end
+
+  test "webpack dev server adds the dev server to Content-Security-Policy on pages triggering a welcome page" do
+    begin
+      Rails.configuration.webpack.dev_server.enabled = true
+      Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+      Environment.stubs(:first).returns(nil)
+      visit '/environments'
+      assert page.response_headers['Content-Security-Policy'].include?(Rails.configuration.webpack.dev_server.port.to_s)
+    ensure
+      Rails.configuration.webpack.dev_server.enabled = false
+    end
+  end
+
+  test "webpack dev server adds the dev server to Content-Security-Policy on unauthirized page requests" do
+    begin
+      Rails.configuration.webpack.dev_server.enabled = true
+      Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+      User.stubs(:current).returns(nil)
```

Comment:
Prefer overriding `login_admin` (see user_test.rb), same comment as above applies - no verification that this tests the `authorize` filter behaviour.